blocks/procedures.js

```diff
@@ -414,132 +435,132 @@ Blockly.Blocks['procedures_defnoreturn'] = {
         argXmlBlock.appendChild(argXmlField);
         argOption.callback =
-            Blockly.ContextMenu.callbackFactory(this, argXmlBlock);
+            ContextMenu.callbackFactory(this, argXmlBlock);
         options.push(argOption);
       }
     }
   },
   callType_: 'procedures_callnoreturn',
 };
 
-Blockly.Blocks['procedures_defreturn'] = {
+Blocks['procedures_defreturn'] = {
   /**
    * Block for defining a procedure with a return value.
-   * @this {Blockly.Block}
+   * @this {Block}
    */
   init: function() {
-    const initName = Blockly.Procedures.findLegalName('', this);
-    const nameField = new Blockly.FieldTextInput(initName,
-        Blockly.Procedures.rename);
+    const initName = Procedures.findLegalName('', this);
+    const nameField = new FieldTextInput(initName,
+        Procedures.rename);
     nameField.setSpellcheck(false);
     this.appendDummyInput()
-        .appendField(Blockly.Msg['PROCEDURES_DEFRETURN_TITLE'])
+        .appendField(Msg['PROCEDURES_DEFRETURN_TITLE'])
         .appendField(nameField, 'NAME')
         .appendField('', 'PARAMS');
     this.appendValueInput('RETURN')
-        .setAlign(Blockly.ALIGN_RIGHT)
-        .appendField(Blockly.Msg['PROCEDURES_DEFRETURN_RETURN']);
-    this.setMutator(new Blockly.Mutator(['procedures_mutatorarg']));
+        .setAlign(Align.RIGHT)
+        .appendField(Msg['PROCEDURES_DEFRETURN_RETURN']);
+    this.setMutator(new Mutator(['procedures_mutatorarg']));
     if ((this.workspace.options.comments ||
          (this.workspace.options.parentWorkspace &&
           this.workspace.options.parentWorkspace.options.comments)) &&
-        Blockly.Msg['PROCEDURES_DEFRETURN_COMMENT']) {
-      this.setCommentText(Blockly.Msg['PROCEDURES_DEFRETURN_COMMENT']);
+        Msg['PROCEDURES_DEFRETURN_COMMENT']) {
+      this.setCommentText(Msg['PROCEDURES_DEFRETURN_COMMENT']);
     }
     this.setStyle('procedure_blocks');
-    this.setTooltip(Blockly.Msg['PROCEDURES_DEFRETURN_TOOLTIP']);
-    this.setHelpUrl(Blockly.Msg['PROCEDURES_DEFRETURN_HELPURL']);
+    this.setTooltip(Msg['PROCEDURES_DEFRETURN_TOOLTIP']);
+    this.setHelpUrl(Msg['PROCEDURES_DEFRETURN_HELPURL']);
     this.arguments_ = [];
     this.argumentVarModels_ = [];
     this.setStatements_(true);
     this.statementConnection_ = null;
   },
-  setStatements_: Blockly.Blocks['procedures_defnoreturn'].setStatements_,
-  updateParams_: Blockly.Blocks['procedures_defnoreturn'].updateParams_,
-  mutationToDom: Blockly.Blocks['procedures_defnoreturn'].mutationToDom,
-  domToMutation: Blockly.Blocks['procedures_defnoreturn'].domToMutation,
-  saveExtraState: Blockly.Blocks['procedures_defnoreturn'].saveExtraState,
-  loadExtraState: Blockly.Blocks['procedures_defnoreturn'].loadExtraState,
-  decompose: Blockly.Blocks['procedures_defnoreturn'].decompose,
-  compose: Blockly.Blocks['procedures_defnoreturn'].compose,
+  setStatements_: Blocks['procedures_defnoreturn'].setStatements_,
```

Comment:
All of these helper functions could actually be defined as file-internal, and then installed on both types of blocks. This structure looks like an attempt to make these functions non-global.

Reply:
Done, by declaring a common object and then doing `{...common, init: …}`.